compiler/tests-compiler/global_deadcode.ml

```diff
@@ -0,0 +1,75 @@
+open Util
+
+let%expect_test "Eliminates unused functions from functor" =
+  let program =
+    (* Create two instances of Set functor so compiler can't inline one implementation. *)
+    compile_and_parse_whole_program
+      {|
+      module Int_set = Set.Make (Int);;
+      module String_set = Set.Make (String);;
+
+      let int_set = Int_set.singleton 1 in
+      let string_set = String_set.empty in
+      let string_set = String_set.add "hello" string_set in
+      print_endline (string_of_int (Int_set.find 1 int_set) ^ (String_set.find "hello" string_set))
+      |}
+  in
+  (* Expect: compiles functor members used in the program *)
+  print_fun_decl program (Some "find");
+  [%expect
+    {|
+    function find(x, param){
+     var param$0 = param;
+     for(;;){
+      if(! param$0) throw caml_maybe_attach_backtrace(Not_found, 1);
+      var
+       r = param$0[3],
+       v = param$0[2],
+       l = param$0[1],
+       c = caml_call2(Ord[1], x, v);
+      if(0 === c) return v;
+      var r$0 = 0 <= c ? r : l, param$0 = r$0;
+     }
+    }
+    //end |}];
+  print_fun_decl program (Some "add");
+  [%expect {|
+    function add(x, t){
+     if(! t) return [0, 0, x, 0, 1];
+     var r = t[3], v = t[2], l = t[1], c = caml_call2(Ord[1], x, v);
+     if(0 === c) return t;
+     if(0 <= c){var rr = add(x, r); return r === rr ? t : bal(l, v, rr);}
+     var ll = add(x, l);
+     return l === ll ? t : bal(ll, v, r);
+    }
+    //end |}];
+  (* Expect: does not compile unused functor member *)
```

Comment:
Would't it be easier to just print `Set.Make` function  ?

Reply:
Probably! Good idea

Reply:
I could do this, but it's a bit fragile since that doesn't have a stable name. It's given the name `_e_` when I run it, but that could change.

Reply:
I've changed the test to first find the name of the functor and then print its body.

There might be a bug to fix later. I would expect this variable to be stable/named ("Make"), @vouillon do you think it can be a missing debug event ?